File: src/main/java/com/dozycoffee/wms/warehouse/domain/valueobject/Address.java
package com.dozycoffee.wms.warehouse.domain.valueobject;

import com.dozycoffee.wms.global.error.InvalidDomainValueException;
import com.dozycoffee.wms.warehouse.domain.exception.WarehouseErrorCode;

public record Address(
        String value
) {
    public Address {
        if (value == null || value.isBlank()) {
            throw new InvalidDomainValueException(WarehouseErrorCode.INVALID_ADDRESS);
        }
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/domain/valueobject/Address.kt
package com.dozycoffee.wms.warehouse.domain.valueobject

import com.dozycoffee.wms.global.error.InvalidDomainValueException
import com.dozycoffee.wms.warehouse.domain.exception.WarehouseErrorCode

@ConsistentCopyVisibility
data class Address private constructor(val value: String) {
    companion object {
        fun of(value: String?): Address {
            if (value.isNullOrBlank()) {
                throw InvalidDomainValueException(WarehouseErrorCode.INVALID_ADDRESS)
            }
            return Address(value)
        }
    }
}
